File: algorithms/1_easy/M/02144_minimum_cost_of_buying_candies-with_discount/solution.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        n = len(cost)
        total_cost = 0

        cost.sort(reverse=True)

        for i in range(0, n, 3):
            total_cost += cost[i] + (cost[i + 1] if i + 1 < n else 0)

        return total_cost

File: algorithms/1_easy/M/02144_minimum_cost_of_buying_candies-with_discount/solution.cpp
class Solution {
public:
    static int minimumCost(vector<int>& cost) {
        const size_t len = cost.size();
        int total_cost = 0;

        std::sort(cost.begin(), cost.end(), std::greater<>());

        for (size_t i = 0; i < len; i += 3) {
            total_cost += cost[i] + (i + 1 < len ? cost[i + 1] : 0);
        }

        return total_cost;
    }
};
